### PR TITLE
Add grafana resource configuration

### LINKED ISSUE #10042: Regression: UI no longer allows for configuration of Grafana requests and limits
Internal reference: SURE-7089
Reported in 2.7.6

Issue description:
The UI no longer allows for the configuration of Grafana limits and requests

Impact:
The user runs a dev cluster, which is imperative to the production of their product and the developers losing access to the metrics on their deployments prevents them from being able to complete necessary tests.  As the fields were not visible, they didn't realize Prometheus configurations didn't apply to all the pods, including Grafana, so what would have been addressed in their initial deployment ended up causing intermittent issues for his users for what may have been months. 

Troubleshooting steps:

After observing that Grafana was using more than its limits allowed, they modified the manifests directly to add better requests and limits.  

Repro steps:

    Edit Monitoring App in the UI

Workaround:
Modify as YAML

Expected behavior:
There used to be fields here for the modification of requests and limits. Since user count and dashboards will constantly change over the span of the Apps years of life in the cluster, this requires periodic tuning.  As such, these are as important part of the configuration of Grafana as Grafana's storage configuration and should be re-exposed to the users.

### PR DESCRIPTION
**Pull request by nwmac**
Fixes #10042

NOTE: This PR is based on top of https://github.com/rancher/dashboard/pull/10085, which needs to be reviewed and merged first.

This PR adds Grafana resource configuration fields to the UI for the Monitoring installation chart:

image

Also added an e2e test to verify that the values are correctly wired in to the relevant backend API request.

### CODE REVIEW

File: shell/chart/monitoring/grafana/index.vue [hunk 1]
```diff
@@ -87,6 +87,9 @@ export default {
 
       return (storageClasses || []).length >= 1;
     },
+    showGrafanaResourceConfig() {
+      return this.value?.grafana?.resources?.requests && this.value?.grafana?.resources?.limits;
```
Comment (richard-cox) [inline shell/chart/monitoring/grafana/index.vue:91]: What happens if these are not present but the user wants to set them? 

The [grafana chart](https://github.com/grafana/helm-charts/tree/main/charts/grafana) states that `resource` by default is an empty object. I'm guessing we have something in `values` though that sets them (components are visible & populated when first going to the page). Would the user ever delete them, then want to add them later in an edit?

**Review by richard-cox — COMMENTED** [review]
One question, otherwise looks good

File: shell/chart/monitoring/grafana/index.vue (hunk 1 above)
Comment (nwmac) [reply shell/chart/monitoring/grafana/index.vue:91]: Yeah - for now, it was easiest to do it this way - our chart should have this. If you delete them then  you won't get the UI. I think that is okay for now.

**Review by richard-cox — APPROVED** [review]

**Comment by nwmac** [comment]
@richard-cox I've updated the e2e tests and Ci is now passing - can you re-review the changes.

**Review by richard-cox — APPROVED** [review]

**Comment by izaac** [comment]
I was checking for negative tests for those fields but it looks that those resource limit input fields doesn't have any validation?.. In that case the coverage looks good to me. @yonasberhe23 please add your review here before moving to Done.

**Pull request merged**